src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs

```diff
@@ -119,69 +102,69 @@ private void Start(string workingDirectory, string fileName, string arguments, I
 
             // Hook up the events.
             _proc.EnableRaisingEvents = true;
-            _proc.Exited += OnExited;
-            if (_proc.StartInfo.RedirectStandardOutput)
-            {
-                _proc.OutputDataReceived += OnOutputDataReceived;
-            }
+            _proc.Exited += ProcessExitedHandler;
 
-            if (_proc.StartInfo.RedirectStandardError)
+            using (var registration = cancellationToken.Register(() => CancelProcessTree()))
             {
-                _proc.ErrorDataReceived += OnErrorDataReceived;
-            }
+                // Start the process.
+                try
+                {
+                    _stopWatch = Stopwatch.StartNew();
+                    _proc.Start();
+                }
+                catch (Exception ex) when (ex is Win32Exception && (ex as Win32Exception)?.NativeErrorCode == 2)
+                {
+                    throw new FileNotFoundException($"File not found: {_proc.StartInfo.FileName}", ex);
+                }
 
-            // Start the process.
-            _stopWatch = Stopwatch.StartNew();
-            bool newProcessStarted = _proc.Start();
-            if (!newProcessStarted)
-            {
-                Trace.Verbose("Used existing process instead of starting new one for " + fileName);
-            }
+                // Close the input stream. This is done to prevent commands from blocking the build waiting for input from the user.
+                _proc.StandardInput.Dispose();
```

Comment:
Is this equivalent to not redirecting STDIN? Is this really required, or was this just copied? If you have more context, it would be nice to leave in a comment in the code with more explanation. I believe some things changed in Net Core, so it would be interesting to repro to see if this is really required.